### PR TITLE
Pentest Issue: remove hardcode credentials

### PR DESCRIPTION
**Pull request by liu-shaojun**
## Description

This submission 
- remove hardcode credential from python/orca/src/bigdl/orca/ray/ray_on_spark_context.py according to PenTest Report Issue [Hardcoded Credential Found in Source Code](https://github.com/analytics-zoo/arda-data/issues/23#:~:text=details%20to%20containers.-,Hardcoded%20Credential%20Found%20in%20Source%20Code,-Affected%20Resources).
  - if users want to specify redis_password, they can add key value into extra_params to launch ray: `extra_params={"redis-password": "xxxxxx"}`
  - if redis_password is not defined in extra_params, the default redis password defined in ray will be used.

### CODE REVIEW

File: python/orca/src/bigdl/orca/ray/ray_on_spark_context.py [hunk 1]
```diff
@@ -329,7 +329,7 @@ def _start_ray_services(iter):
 class RayOnSparkContext(object):
     _active_ray_context = None
 
-    def __init__(self, sc, redis_port=None, password="123456", object_store_memory=None,
+    def __init__(self, sc, redis_port=None, password=None, object_store_memory=None,
```
Comment (sgwhat) [inline python/orca/src/bigdl/orca/ray/ray_on_spark_context.py:332]: In line 349, change the default value to None. 😄

**Review by sgwhat — COMMENTED** [review]
Need to run ut.

File: python/orca/src/bigdl/orca/ray/ray_on_spark_context.py (hunk 1 above)
Comment (liu-shaojun) [reply python/orca/src/bigdl/orca/ray/ray_on_spark_context.py:332]: Good catch! updated! thx!

Comment (glorysdj) [reply python/orca/src/bigdl/orca/ray/ray_on_spark_context.py:332]: None and "None" are different

**Review by hkvision — APPROVED** [review]

**Comment by hkvision** [comment]
Need to run on yarn or k8s to verify the change :)

File: python/orca/src/bigdl/orca/ray/ray_on_spark_context.py [hunk 2]
```diff
@@ -546,19 +547,17 @@ def init(self, driver_cores=0):
...
                     num_cpus=self.ray_node_cpu_cores,
-                    _redis_password=self.redis_password,
                     object_store_memory=self.object_store_memory,
                     include_dashboard=self.include_webui,
                     dashboard_host="0.0.0.0",
                     _system_config=self.system_config
                 )
+                if "redis_password" in kwargs:
+                    kwargs["_redis_password"] = kwargs["redis_password"]
+                    kwargs.pop("redis_password")
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/ray/ray_on_spark_context.py:560]: Can put this part into `update_extra_params`?

Comment (liu-shaojun) [reply python/orca/src/bigdl/orca/ray/ray_on_spark_context.py]: sure, updated!

**Comment by liu-shaojun** [comment]
The code changes have been verified on yarn and k8s.

**Pull request merged**